tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -211,6 +211,69 @@ class Restrictions {
     return [];
   }
 
+  List<SourceSpanSeverityException> validateExtendsClassKey(
+    String parentNodeName,
+    dynamic extendsClass,
+    SourceSpan? span,
+  ) {
+    if (extendsClass is! String) {
+      return [
+        SourceSpanSeverityException(
+          'The "extends" key must be a String.',
+          span,
+        )
+      ];
+    }
```

Comment:
@BenAuerDev check how i did for defaults https://github.com/serverpod/serverpod/blob/main/tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions/default.dart

i know it may not be the same structure and logic but use it as reference

Reply:
@klkucaj thanks :rocket: 

@SandPod `documentType` represents: 

> The [documentType] represents the parent key of the [documentContents],
in the initial processing this is expected to be the top level model type
we are checking. E.g. 'class', 'enum', 'exception', etc.

Using `documentType` would results in:

![error-from-doc-type](https://github.com/user-attachments/assets/fa704ef1-cfea-40a1-8750-09ed2ddf96f9)

compared to before:

![error-from-property-name](https://github.com/user-attachments/assets/aae66a26-c90f-4dfa-9afe-4cb8e9b570b0)


Which **kind of** makes sense as the value of  `extendsClass`  should be a class but I'm not 100% sure. Mostly because it might be a little harder to understand when the error references `class` type but points to `extends`. 

 What do you think? 

One thing that I just noticed, is that I used "property" where I should have used "type" in the error message

Reply:
I was incorrect here, your implementation is correct! :)